### PR TITLE
Replace home-brewn eventbus with Forge's eventbus.

### PR DESCRIPTION
**Pull request by magicus**
EventSubclassTransformer still not working properly for Fabric.

### CODE REVIEW

File: common/src/main/java/com/wynntils/framework/feature/Feature.java [hunk 1]
```diff
@@ -9,12 +9,14 @@
 public abstract class Feature {
     /** Called on a feature's activation */
     public void onEnable() {
-        WynntilsMod.eventBus.register(this);
+        WynntilsMod.EVENT_BUS.register(this);
+        WynntilsMod.EVENT_BUS.register(this.getClass());
```
Comment (Incompleteusern) [inline common/src/main/java/com/wynntils/framework/feature/Feature.java:13]: Is there a case where both static and nonstatic are used?

Comment (magicus) [reply common/src/main/java/com/wynntils/framework/feature/Feature.java:13]: Right now, no. (It was before I rebased it on your latest changes :))

Comment (magicus) [reply common/src/main/java/com/wynntils/framework/feature/Feature.java:13]: But if the registration should be done by the abstract superclass, I think we should cater to both static and instance listeners.

File: common/src/main/java/com/wynntils/mc/event/InventoryRenderEvent.java [hunk 2]
```diff
@@ -5,17 +5,20 @@
...
-    private final Slot hoveredSlot;
+    private Screen screen;
+    private PoseStack poseStack;
+    private int mouseX;
+    private int mouseY;
+    private float partialTicks;
+    private Slot hoveredSlot;
+
+    public InventoryRenderEvent() {
+    }
```
Comment (magicus) [inline common/src/main/java/com/wynntils/mc/event/InventoryRenderEvent.java:22]: This change is fundamentally broken, and can hopefully be reverted if I can get the constructor transformer thingie to work on fabric.

File: common/src/main/java/com/wynntils/mc/event/TitleScreenInitEvent.java [hunk 3]
```diff
@@ -4,16 +4,19 @@
...
 import java.util.function.Consumer;
 import net.minecraft.client.gui.components.AbstractWidget;
 import net.minecraft.client.gui.screens.TitleScreen;
+import net.minecraftforge.eventbus.api.Event;
 
 public class TitleScreenInitEvent extends Event {
-    private final TitleScreen titleScreen;
-    private final List<AbstractWidget> buttons;
-    private final Consumer<AbstractWidget> addButton;
+    private TitleScreen titleScreen;
```
Comment (magicus) [inline common/src/main/java/com/wynntils/mc/event/TitleScreenInitEvent.java:14]: ... as is this.

**Comment by magicus** [comment]
I believewe need to call `processClass` in `net.minecraftforge.eventbus.EventBusEngine` to process the event classes, in fabric. (This is done automatically in forge). I suspect that unless we manage to do this, the fabric version of the event bus will be just as inefficient as the home-brewn version...

**Comment by Incompleteusern** [comment]
It will be definitely more efficient due to usage of asm, but the transformer will need to be called. Im curious how they are able to find all the relevant classes, is it something to do with ClassLoader

**Pull request merged**